### ISSUE TITLE
Real-life Example not working

### ISSUE THREAD

**Issue by smooJitter** [wix/react-native-interactable#123]
![simulator screen shot jul 21 2017 9 51 26 am](https://user-images.githubusercontent.com/13536334/28473648-68cbd694-6dfa-11e7-9a6a-20603d4a7bbc.png)

The animations are not working. Environment:
```
    "mantra-core": "^1.7.0",
    "native-base": "^2.1.5",
    "react": "16.0.0-alpha.6",
    "react-native": "0.44.3",
    "react-native-app-intro": "^1.1.5",
    "react-native-clean-form": "^0.4.2",
    "react-native-easy-grid": "^0.1.13",
    "react-native-elements": "^0.13.0",
    "react-native-fit-image": "^1.4.8",
    "react-native-interactable": "^0.1.0",
    "react-native-linear-gradient": "^2.0.0",
    "react-native-modal-dropdown": "^0.4.4",
    "react-native-multi-slider": "^0.3.5",
    "react-native-swiper": "^1.5.4",
    "react-native-vector-icons": "^4.2.0",
    "react-redux": "^5.0.5",
    "redux": "^3.7.2",
    "redux-form": "^6.8.0"```

**Comment by Horizon-Blue** [comment]
Have you [link the library to your project](https://github.com/wix/react-native-interactable#installation)? If not, try
```bash
react-native link react-native-interactable
```
then rebuild the app

**Comment by chrismllr** [comment]
I am also seeing this, but when installing `react-native-interactable` on a fresh test project.

I've run `react-native link react-native-interactable` and verified it:
![screen shot 2017-08-15 at 6 25 55 pm](https://user-images.githubusercontent.com/9942917/29340929-37ea563a-81e7-11e7-8eb2-7c5221d85b56.png)

Versions:
```
"react": "16.0.0-alpha.12",
"react-native": "0.47.1",
"react-native-interactable": "^0.1.1",
```

Is it possible this is an issue caused by react 16 alpha?

**Comment by chrismllr** [comment]
This is the console error I'm seeing. It really does feel like an error with the link, imo

![screen shot 2017-08-15 at 6 28 41 pm](https://user-images.githubusercontent.com/9942917/29341036-a66bca76-81e7-11e7-8680-3b7d549eed27.png)

**Comment by tangtai** [comment]
I'm having the same problem were you able to solve it chris?

**Comment by AaronBuxbaum** [comment]
This might be a problem with React 16 (the beta, at least), but many other libraries require it. Until this bug is fixed, I can't use it :(

**Comment by luugiathuy** [comment]
@smooJitter @chrismllr @tangtai @AaronBuxbaum This commit fixes your issue https://github.com/wix/react-native-interactable/commit/d8c8f08f2b805070eac333124149ce87132a986e. For now, while waiting for the repo to update new version to include this commit. You can use this commit in your project by define the `react-native-interactable` under `dependencies` in your `package.json` as follow:

```
"react-native-interactable": "https://github.com/wix/react-native-interactable.git#d8c8f08f2b805070eac333124149ce87132a986e"
```

Hope it helps!

**Comment by AaronBuxbaum** [comment]
@luugiathuy thanks! but it doesn't fix my problem.

```
11:18:16 AM: Warning: 'react-native' peer depencency missing. Run `npm ls` in ... to see full warning.
Warning: Installed version 0.1.1 of 'react-native-interactable' does not satisfy required version https://github.com/wix/react-native-interactable.git#d8c8f08f2b805070eac333124149ce87132a986e

If there is an issue running your project, please run `npm install` in ... and restart.

11:18:25 AM: Finished building JavaScript bundle in 3431ms
11:18:25 AM: Warning: Native component for "InteractableView" does not exist

11:18:25 AM: Warning: checkPropTypes has been moved to a separate package. Accessing React.checkPropTypes is no longer supported and will be removed completely in React 16. Use the prop-types package on npm instead. (https://fb.me/migrating-from-react-proptypes)
```

Warnings could be from other packages, but it has the same "component does not exist" bug in the end.

**Comment by luugiathuy** [comment]
@AaronBuxbaum Have you tried to run `react-native link react-native-interactable` again?

**Comment by AaronBuxbaum** [comment]
@luugiathuy yup, also deleted and reinstalled node modules just in case -- same error.

**Comment by AaronBuxbaum** [comment]
I should mention that I'm using CRNA, that might be relevant to troubleshooting

**Comment by AaronBuxbaum** [comment]
@luugiathuy confirmed that this doesn't fix the problem. I built a new empty project with CRNA, tested that it was working, then attempted to add `react-native-interactable` with the above URL. 

`package.json` dependencies:
```
    "expo": "^20.0.4",
    "react": "16.0.0-alpha.12",
    "react-native": "^0.47.2",
    "react-native-interactable": "https://github.com/wix/react-native-interactable.git#d8c8f08f2b805070eac333124149ce87132a986e"
```

Logs:
```
4:36:35 PM: Warning: Installed version 0.1.1 of 'react-native-interactable' does not satisfy required version https://github.com/wix/react-native-interactable.git#d8c8f08f2b805070eac333124149ce87132a986e
4:37:34 PM: Finished building JavaScript bundle in 1198ms
4:37:34 PM: Warning: Native component for "InteractableView" does not exist
```

Simulator:
![screen shot 2017-08-30 at 4 38 46 pm](https://user-images.githubusercontent.com/5578581/29893981-eb704c8a-8da1-11e7-88e5-e36426a44479.png)

`npm ls`:
```
npm ERR! peer dep missing: react@>= 15.4.1, required by react-native-interactable@0.1.1
npm ERR! peer dep missing: react@>=15.3.1, required by lottie-react-native@1.1.1
npm ERR! peer dep missing: react@>=15.4.0, required by react-native-branch@2.0.0-beta.3
npm ERR! peer dep missing: react@>= 16.0.0, required by react-native-gesture-handler@1.0.0-alpha.
```

**Comment by pcooney10** [comment]
@AaronBuxbaum CRNA& Expo are not compatible with any projects that use `react-native link`

**Comment by Foberm** [comment]
@pcooney10 is there a way to preview my code without building the project everytime if crna and expo are not working?

**Comment by pcooney10** [comment]
I am not sure, I currently only develop with Expo and CRNA.

For more information on `react-native link`, you can look [here](https://docs.expo.io/versions/latest/guides/expokit.html#changing-native-dependencies).

**Comment by smith64fx** [comment]
I just cloned the current git repo and it worked fine :)

The important thing is: You have to use the right version of react / react-native in your project!

This issue can be closed!

**Comment by ArtiomShapovalov** [comment]
Manual linking helped. Check it out [here](http://facebook.github.io/react-native/docs/linking-libraries-ios.html#manual-linking)

**Comment by ngocnguyen09910060** [comment]
I still get same error. Please help me.
This is my package.json:
`"dependencies": {
    "react": "^16.2.0",
    "react-native": "0.52.0",
    "react-native-interactable": "^0.1.7",
    "react-native-vector-icons": "^4.5.0",
    "react-navigation": "^1.0.0-beta.27"
  },`

**Comment by MariusMeiners** [comment]
same issue... i guess its not compatible with react 16.2.0 at this current point in time...

**Comment by Thebinoman** [comment]
Same issue.. not working

**Comment by traxx10** [comment]
Same issue, Not working too

**Comment by TheGreatNectarine** [comment]
@luugiathuy YOU SAVED MY DAY MAN!!! Thank you soooo muuuch!!!

**Comment by andyesp** [comment]
If someone is having the `Native component for "InteractableView" does not exist` error, doing `react-native link react-native-interactable` is not enough. You have to add the module manually to the `MainApplication.java`. Check that file in the real-life-example source code.
